samples/drivers/rtc/src/main.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2024, Muhammad Waleed Badar
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/rtc.h>
+#include <zephyr/sys/util.h>
+
+const struct device *const rtc = DEVICE_DT_GET(DT_ALIAS(rtc));
+
+#ifdef SET_DATE_TIME
+static int set_date_time(const struct device *rtc)
+{
+	int ret = 0;
+	struct rtc_time tm = {.tm_year = 2024 - 1900,
+			      .tm_mon = 7 - 1,
+			      .tm_mday = 28,
+			      .tm_hour = 3,
+			      .tm_min = 51,
+			      .tm_sec = 0};
+
+	ret = rtc_set_time(rtc, &tm);
+	if (ret < 0) {
+		printk("Cannot write date time: %d\n", ret);
+		return ret;
+	}
+	return ret;
+}
+#endif
+
+static int get_date_time(const struct device *rtc)
+{
+	int ret = 0;
+	struct rtc_time tm;
+
+	ret = rtc_get_time(rtc, &tm);
+	if (ret < 0) {
+		printk("Cannot read date time: %d\n", ret);
+		return ret;
+	}
+
+	printk("Current RTC time: %04d-%02d-%02d %02d:%02d:%02d\n", tm.tm_year + 1900,
+	       tm.tm_mon + 1, tm.tm_mday, tm.tm_hour, tm.tm_min, tm.tm_sec);
+
+	return ret;
+}
+
+int main(void)
+{
+	/* Check if the RTC is ready */
+	if (!device_is_ready(rtc)) {
+		printk("Device is not ready\n");
+		return 0;
+	}
+
+#ifdef SET_DATE_TIME
+	set_date_time(rtc);
+#endif
+
+	/* Continuously read the current date and time from the RTC */
+	while (get_date_time(rtc) == 0) {
+		if (get_date_time(rtc)) {
+			printk("Failed to read from rtc\n");
+			return 0;
+		}
+		k_sleep(K_MSEC(1000));
+	};
+	return 0;
```

Comment:
The helper `get_date_time()` already prints that it failed to get the time, including the error code, so adding an additional `get_date_time()` call and printing `printk("Failed to read from rtc\n");` is redundant IMO.

It would produce:
```
Current RTC time: ...
Cannot read date time: -1
Failed to read from rtc
```

Reply:
Additionally, it can then be simplified to:
```
while (get_date_time(rtc) == 0) {
        k_sleep(K_MSEC(1000));
}
```

Reply:
Simplified the logic as you suggested. I appreciate your feedback and hope that this pull request is not taking too much of your time.